test/vecelement.jl

```diff
@@ -0,0 +1,19 @@
+# This file is a part of Julia. License is MIT: http://julialang.org/license
+
+typealias Vec{N,T} NTuple{N,Base.VecElement{T}}
+
+# Crash report for #15244 motivated this test.
+@generated function thrice_iota{N,T}(::Type{Vec{N,T}})
+    Expr(:tuple, [:(Base.VecElement(3*$i)) for i in 1:N]...)
+end
+
+function call_iota(N)
+    x = thrice_iota(Vec{N,Float32})
+    @assert x[1].value+x[N].value == 3+3*N
```

Comment:
Why not use `@test` here?

You can also compare with `===`, so that type equality is also checked.